seqr/utils/search/add_data_utils.py

```diff
@@ -3,14 +3,73 @@
 from seqr.models import Sample
 from seqr.utils.search.constants import VCF_FILE_EXTENSIONS
 from seqr.utils.search.utils import get_es_client, get_index_metadata
+from seqr.views.utils.dataset_utils import match_and_update_search_samples, load_mapping_file
 
 
 SAMPLE_FIELDS_LIST = ['samples', 'samples_num_alt_1']
 #  support .bgz instead of requiring .vcf.bgz due to issues with DSP delivery of large callsets
 DATASET_FILE_EXTENSIONS = VCF_FILE_EXTENSIONS[:-1] + ('.bgz', '.bed', '.mt')
 
 
-def validate_index_metadata_and_get_samples(elasticsearch_index, **kwargs):
+def add_new_search_samples(request_json, project, user, summary_template=False, expected_families=None):
+    required_fields = ['elasticsearchIndex', 'datasetType']
+    if any(field not in request_json for field in required_fields):
+        raise ValueError(f'request must contain fields: {", ".join(required_fields)}')
+
+    dataset_type = request_json['datasetType']
+    if dataset_type not in Sample.DATASET_TYPE_LOOKUP:
+        raise ValueError(f'Invalid dataset type "{dataset_type}"')
+
+    elasticsearch_index = request_json['elasticsearchIndex'].strip()
+    dataset_type = request_json['datasetType']
+    ignore_extra_samples = request_json.get('ignoreExtraSamplesInCallset')
+    genome_version = request_json.get('genomeVersion')
+    sample_id_to_individual_id_mapping = load_mapping_file(
+        request_json['mappingFilePath'], user) if request_json.get('mappingFilePath') else {}
+
+    sample_ids, sample_type = _validate_index_metadata_and_get_samples(
+        elasticsearch_index, project=project, dataset_type=dataset_type, genome_version=genome_version)
+    if not sample_ids:
+        raise ValueError('No samples found in the index. Make sure the specified caller type is correct')
+
+    sample_data = {
+        'elasticsearch_index': elasticsearch_index,
+    }
+    num_samples, matched_individual_ids, activated_sample_guids, inactivated_sample_guids, updated_family_guids = match_and_update_search_samples(
+        project=project,
+        user=user,
+        sample_ids=sample_ids,
+        sample_data=sample_data,
```

Comment:
Are there any benefits to replacing the `elasticsearch_index` with `sample_data`?

Reply:
yeah we are gonna replace this with a different model field for the new backend and it will be good to be able to use the existing helper function but have all the backend-specific logic in one place